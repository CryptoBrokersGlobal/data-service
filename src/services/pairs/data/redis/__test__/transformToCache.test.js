const transformToCache = require('../transformToCache');
const { BigNumber } = require('@waves/data-entities');
const pair = require('../../__test__/mocks/pair');

describe('Redis transformToCache', () => {
  it('should do nothing to an empty array', () => {
    expect(transformToCache([])).toEqual([]);
  });

  it('should correctly transform a valid toCache array', () => {
    const p = pair('WAVES', 'BTC');
    expect(
      transformToCache([[p, { q: new BigNumber(20000) }]])
    ).toMatchSnapshot();
  });
});
