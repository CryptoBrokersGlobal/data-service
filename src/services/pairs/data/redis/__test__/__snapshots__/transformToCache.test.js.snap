// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redis transformToCache should correctly transform a valid toCache array 1`] = `
Array [
  "pair:WAVES/8LQW8f7P5d5PZM7GtZEBgaqRPGSzS3DfPuiXrURJ4AJS",
  "{\\"q\\":20000}",
]
`;
